app/views/v2/no_results.scala.html

```diff
@@ -1,26 +1,28 @@
 
 @import uk.gov.hmrc.play.views.html.helpers
 @import views.html.v2.main_template
-
+@import model.MessageConstants.{EnglishMessageConstants, WelshMessageConstants, MessageConstants}
 @import model._
 
-@(id: String, journeyData: JourneyDataV2, postcode: String)(implicit request: Request[_], messages: Messages)
+@(id: String, journeyData: JourneyDataV2, postcode: String, isWelsh: Boolean = false)(implicit request: Request[_], messages: Messages)
+
+@messageConstants = @{if(isWelsh) WelshMessageConstants else EnglishMessageConstants}
 
-@main_template(title = messages("result.title.text"), journeyData = Some(journeyData)) {
+@main_template(title = messageConstants.noResultsPageTitle, journeyData = Some(journeyData)) {
```

Comment:
When calling the main template, you need to pass the welsh flag through to it.

Reply:
OK, I'll change this